src/ssd/ssd.c

```diff
@@ -267,6 +262,25 @@ ssd_update_geometry(struct ssd *ssd)
 		}
 		return;
 	}
+
+	if (current.width < LAB_MIN_VIEW_WIDTH) {
+		/*
+		 * This draws a border-colored background behind the
+		 * view for poorly behaving clients that don't follow
+		 * our requested minimal size.
+		 */
+		if (!ssd->view_background) {
+			ssd->view_background = wlr_scene_rect_create(ssd->tree, 0, 0,
+				ssd->view == ssd->view->server->active_view
+					? rc.theme->window_active_border_color
+					: rc.theme->window_inactive_border_color);
+		}
+		wlr_scene_rect_set_size(ssd->view_background, LAB_MIN_VIEW_WIDTH, current.height);
```

Comment:
Maybe we should resize the background to the size of padding area rather than the whole view size, because the surface may be translucent.